tests/unit/package_managers/bundler/test_main.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from git.repo import Repo
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.package_managers.bundler.main import _get_main_package_name_and_version
+from cachi2.core.package_managers.bundler.parser import GemDependency, ParseResult, PathDependency
+from cachi2.core.rooted_path import RootedPath
+
+
+def test_get_main_package_name_and_version(rooted_tmp_path: RootedPath) -> None:
+    dependencies: ParseResult = [
+        GemDependency(
+            name="my_gem_dep",
+            version="0.1.0",
+            source="https://rubygems.org",
+        ),
+        PathDependency(
+            name="my_path_dep",
+            version="0.2.0",
+            path=".",
+        ),
+    ]
+
+    name, version = _get_main_package_name_and_version(
+        package_dir=rooted_tmp_path, dependencies=dependencies
+    )
+    assert name == "my_path_dep"
+    assert version == "0.2.0"
+
+
+def test_get_main_package_name_and_version_from_repo(
+    rooted_tmp_path_repo: RootedPath,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    repo = Repo(rooted_tmp_path_repo)
+    repo.create_remote("origin", "git@github.com:user/example.git")
+
+    name, version = _get_main_package_name_and_version(
+        package_dir=rooted_tmp_path_repo, dependencies=[]
+    )
+    assert "Failed to infer name and version from gemspec file" in caplog.messages
+
+    assert name == "example"
+    assert version is None
+
+
+def test_get_main_package_name_and_version_from_repo_without_origin(
+    rooted_tmp_path_repo: RootedPath,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    with pytest.raises(PackageRejected) as exc_info:
+        _get_main_package_name_and_version(package_dir=rooted_tmp_path_repo, dependencies=[])
+
+    assert "Failed to infer name and version from gemspec file" in caplog.messages
+    assert "Failed to infer package name from origin remote" in exc_info.value.friendly_msg()
+
+
+@pytest.mark.skip(reason="Not implemented in `cachi2.core.scm.get_repo_id`")
+def test_get_main_package_name_and_version_from_non_repo(
+    rooted_tmp_path: RootedPath,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    with pytest.raises(PackageRejected) as exc_info:
+        _get_main_package_name_and_version(package_dir=rooted_tmp_path, dependencies=[])
+
+    assert "Failed to infer name and version from gemspec file" in caplog.messages
+    assert "Failed to infer package name from origin remote" in exc_info.value.friendly_msg()
```

Comment:
I guess adding a skipped test like this will get easily forgotten the same way as if it weren't even there, meaning that it's unlikely anyone will remember to enable this test when the condition no longer applies, so I'd argue it's as good as dropping it for now.

Reply:
> Would it help to link the issue I created yesterday to this code?

Probably yes. It will still be forgotten unless you do it yourself, however, there may be a chance that if you do
```
TODO: Enable this test once XYZ is fixed
```
someone might accidentally stumble upon the URL and see if anyone has already done something about that :) .

Reply:
> unless you do it yourself

That is a very likely scenario :)